code/modules/projectiles/guns/ballistic.dm

```diff
@@ -60,8 +61,9 @@
 				magazine = AM
 				to_chat(user, "<span class='notice'>You load a new magazine into \the [src].</span>")
 				chamber_round()
-				A.update_icon()
-				update_icon()
+				if(!dont_update)
+					A.update_icon()
```

Comment:
why do you need to exclude the ammo box from updating its icon? There's no unique ammo boxes here...likewise, even if there was, you could just click this on some other gun, and it would update.